lib/rules/no-confusing-arrow.js

```diff
@@ -65,7 +67,7 @@ module.exports = {
         function checkArrowFunc(node) {
             const body = node.body;
 
-            if (isConditional(body) && !(allowParens && astUtils.isParenthesised(sourceCode, body))) {
+            if (isConditional(body) && (!(allowParens && astUtils.isParenthesised(sourceCode, body)) || (onlyOneSimpleParam && node.params.length === 1 && node.params[0].type !== "ArrayPattern" && node.params[0].type !== "ObjectPattern"))) {
```

Comment:
```suggestion
            if (
                isConditional(body) &&
                !(allowParens && astUtils.isParenthesised(sourceCode, body)) &&
                !(onlyOneSimpleParam && !(node.params.length === 1 && node.params[0].type === "Identifier"))
            ) {
```

The condition was missing `AssignmentPattern` and `RestElement`, which can appear in `params`. It seems best to just check if the param is not `Identifier`.

Also, the condition was logically incorrect, it was allowing arrow functions with one simple param and reporting others instead of the opposite.